src/services/ElasticsearchService.js

```diff
@@ -26,10 +26,13 @@ class ElasticsearchService {
         break;
     }
 
-    this.client = new ElasticsearchClient({
-      ...this.options,
-      Connection,
-    });
+    this.client =
+      typeof client === 'undefined' || client === null || client === false
+        ? new ElasticsearchClient({
```

Comment:
SearchMicroserviceClient can still exist and extend ElasticsearchClient but it will need to have it's own service and utils so e.g. currently we have 

ElasticsearchClient > ElasticSearchService > elasticsearch

You can use the same format and since we customized the client itself then that means we use a different set of object rules

SearchMicroserviceClient > SearchMicroserviceClientService > createElasticSearchService